src/evict/evict_page.c

```diff
@@ -295,7 +301,10 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
         conn->cache->evict_max_page_size = page->memory_footprint;
 
     /* Figure out whether reconciliation was done on the page */
-    clean_page = __wt_page_evict_clean(page);
+    if (__wt_page_evict_clean(page)) {
+        clean_page = true;
```

Comment:
I can remove this bool value but I think we should not use the stats flags to control the flow here. The stats flags should be dedicated to stats logic. Therefore, I still keep it here.

Reply:
I agree that it makes sense to leave it.